### PR TITLE
Document SAML assertion encryption as SP

### PR DESCRIPTION
**Pull request by spwitt**
API and IdP doc updates for SAML assertion encryption when FusionAuth is the SAML SP.

Also reorders fields for UI documentation to match updated screenshots and adds documentation for fields that were missing from admin UI documentation.

### CODE REVIEW

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/enterprise/samlv2-idp-initiated.mdx [hunk 1]
```diff
@@ -70,35 +70,50 @@ This will take you to the <strong>Add SAML v2 IdP Initiated</strong> panel. Here
...
+    The name of the email claim returned in the SAML response.
 
-    If you are running into integration issues, toggle this setting on and FusionAuth will output debugging information into the Event Log during a SAML login. You can find the event log in <strong>System -> Event Log</strong>.
+    When <InlineField>Use NameId for email</InlineField> is enabled this field will not be displayed and will not be required.
+  </APIField>
+  <APIField name="Username claim" optional>
+    The name of the username claim returned in the SAML response.
+  </APIField>
+  <APIField name="Require encrypted assertions" optional since="1.55.1">
+    When enabled FusionAuth requires encrypted assertions in SAML responses from the identity provider. SAML responses from the identity provider containing unencrypted assertions will be rejected by FusionAuth.
```
Comment (hjaret) [inline astro/src/content/docs/lifecycle/authenticate-users/identity-providers/enterprise/samlv2-idp-initiated.mdx:113]: I would personally put a comma after `When enabled` but it's so inconsistent across the documentation that this one doesn't really matter.

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/enterprise/samlv2-idp-initiated.mdx [hunk 2]
```diff
@@ -70,35 +70,50 @@ This will take you to the <strong>Add SAML v2 IdP Initiated</strong> panel. Here
...
-  <APIField name="Debug" optional>
-    Some identity providers are not compliant with the SAML and XML signing specifications. This makes it challenging to get them working with FusionAuth.
+  <APIField name="Use NameId for email" optional>
+    If this is enabled, FusionAuth will assume that the `NameID` in the SAML response contains the email address of the user.
+  </APIField>
+  <APIField name="Email claim" required>
+    The name of the email claim returned in the SAML response.
 
-    If you are running into integration issues, toggle this setting on and FusionAuth will output debugging information into the Event Log during a SAML login. You can find the event log in <strong>System -> Event Log</strong>.
+    When <InlineField>Use NameId for email</InlineField> is enabled this field will not be displayed and will not be required.
```
Comment (hjaret) [inline astro/src/content/docs/lifecycle/authenticate-users/identity-providers/enterprise/samlv2-idp-initiated.mdx:107]: I would personally put a comma after `When <InlineField>Use NameId for email</InlineField> is enabled` but commas after sentences starting with `when` is very inconsistent across the documentation unlike `if` which almost always gets a comma.

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/overview-samlv2.mdx [hunk 3]
```diff
@@ -149,6 +152,12 @@ This will take you to the `Add SAML v2` screen. Here you will need to fill out t
   <APIField name="Alternate destinations" optional since="1.43.0">
     The array of URLs that FusionAuth will accept as SAML login destinations if the <InlineField>Destination Assertion Policy</InlineField> is `Allow alternates`.
   </APIField>
+  <APIField name="Require encrypted assertions" optional since="1.55.1">
+    When enabled FusionAuth requires encrypted assertions in SAML responses from the identity provider. SAML responses from the identity provider containing unencrypted assertions will be rejected by FusionAuth.
```
Comment (hjaret) [inline astro/src/content/docs/lifecycle/authenticate-users/identity-providers/overview-samlv2.mdx:156]: Another `when`.  I'll stop pointing these out now.

**Review by hjaret — DISMISSED** [review]
A couple of subjective punctuation comments.

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/overview-samlv2.mdx [hunk 4]
```diff
@@ -149,6 +152,12 @@ This will take you to the `Add SAML v2` screen. Here you will need to fill out t
...
+  </APIField>
+  <APIField name="Assertion decryption key" optional since="1.55.1">
```
[the remark this answers is not in the document]
Comment (spwitt) [reply astro/src/content/docs/lifecycle/authenticate-users/identity-providers/overview-samlv2.mdx:158]: Added this info for regular and IdP-init

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/overview-samlv2.mdx [hunk 5]
```diff
@@ -119,14 +119,17 @@ This will take you to the `Add SAML v2` screen. Here you will need to fill out t
...
+  </APIField>
   <APIField name="Use POST method" optional>
     When enabled the authentication request will use the HTTP POST binding with the identity provider instead of the default Redirect binding which uses the HTTP GET method.
   </APIField>
   <APIField name="Sign requests" optional>
     When enabled authentication requests sent to the Identity Provider will be signed.
   </APIField>
   <APIField name="Request signing key" optional>
-    The key used to sign the SAML request. Required when <InlineField>Sign request</InlineField> is enabled. To create, manage or import a key, navigate to <strong>Settings -> Key Master</strong>.
+    The key used to sign the SAML request. Required when <InlineField>Sign request</InlineField> is enabled. To create, manage or import a key, navigate to **Settings -> Key Master**.
```
[the remark this answers is not in the document]
Comment (spwitt) [reply astro/src/content/docs/lifecycle/authenticate-users/identity-providers/overview-samlv2.mdx]: The requirements for this key are slightly more complex, and I'm less confident in succinctly describing what's needed. Going to leave this for now.

**Review by andrewpai — APPROVED** [review]

**Pull request merged**